tools/refguide_check.py

```diff
@@ -655,11 +671,18 @@ def flush(self):
         # try to ensure random seed is NOT reproducible
         np.random.seed(None)
 
+        ns = {}
         for t in tests:
+            # We broke the tests up into chunks to try to avoid PSEUDOCODE
+            # This has the unfortunate side effect of restarting the global
+            # namespace for each test chunk, so variables will be "lost" after
+            # a chunk. Chain the globals to avoid this
+            t.globs.update(ns)
```

Comment:
Can you give an example of where we failed without this chaining?

Reply:
On [user.genfromtxt](https://numpy.org/devdocs/user/basics.io.genfromtxt.html?highlight=genfromtxt#tweaking-the-conversion), int the `Converted case ...` block which is a continuation of the block above without repeating the `data` and `convertfunc` definitions